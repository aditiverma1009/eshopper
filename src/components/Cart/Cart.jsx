import React, { Component } from "react";
import "./Cart.css";
export default class Cart extends Component {
  render() {
    if (this.props.cart.length > 0) {
      return (
        <div className="cart-page">
          <div>Cart</div>
          <div>
            {this.props.cart.map((eachCartItem) => {
              return (
                <div key={eachCartItem.id}>
                  <p>{eachCartItem.name}</p>
                  <p>Rs. {eachCartItem.price}</p>
                  <p>Quantity: {eachCartItem.cart}</p>
                </div>
              );
            })}
          </div>
        </div>
      );
    } else {
      return <div className="cart-page">No Items in the cart</div>;
    }
  }
}
