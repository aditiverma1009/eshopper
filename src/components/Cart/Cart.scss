.cart-page {
  padding: 10px;
  .cart-table {
    border-collapse: collapse;
    margin: auto;

    td,
    th,
    tr {
      border: 1px solid grey;
      text-align: left;
      padding: 8px;
    }
  }
  .total-value {
    display: flex;
    justify-content: center;
    flex-direction: column;
    div,
    button {
      margin: 5px;
    }
  }
}
