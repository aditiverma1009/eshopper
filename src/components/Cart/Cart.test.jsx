import { render, waitFor, fireEvent } from "@testing-library/react";
import Cart from "./Cart.jsx";
import React from "react";
import axiosMock from "axios";

const mockCart = [
  {
    id: 1,
    name: "apple",
    price: 120,
    count: 1,
    category: "Fruits & Vegatables",
    inCartCount: 1,
  },
];

const mockPostData = {
  items: [
    {
      id: 1,
      name: "apple",
      price: 120,
      count: 1,
      category: "Fruits & Vegatables",
    },
  ],
};

const mockResponse = {
  data: {
    items: [
      {
        id: 1,
        name: "apple",
        price: 120,
        count: 1,
        category: "Fruits & Vegatables",
      },
    ],
    id: 5,
    date: 1615275152193,
  },
};

describe("Cart", () => {
  test("should render", async () => {
    axiosMock.post = jest.fn().mockResolvedValueOnce(mockResponse);
    const { getByText, container } = render(<Cart cart={mockCart} />);
    const result = await waitFor(() => getByText("Item"));
    expect(result).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("should post cart on checkout", async () => {
    axiosMock.post.mockResolvedValueOnce(mockResponse);
    const { getByText, container } = render(<Cart cart={mockCart} />);
    const button = getByText("Checkout");
    fireEvent.click(button);
    expect(axiosMock.post).toHaveBeenCalledTimes(1);
    expect(axiosMock.post).toHaveBeenCalledWith("/orders", mockPostData, {
      headers: {
        "Content-Type": "application/json",
      },
    });
    expect(container).toMatchSnapshot();
  });
});
