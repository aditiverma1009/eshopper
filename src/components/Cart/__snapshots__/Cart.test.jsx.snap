// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should post cart on checkout 1`] = `
<div>
  <div
    class="cart-page"
  >
    <table
      class="cart-table"
    >
      <tr>
        <td>
          Item
        </td>
        <td>
          Unit Price
        </td>
        <td>
          Quatity
        </td>
        <td>
          Sub Total
        </td>
      </tr>
      <tr>
        <td>
          apple
        </td>
        <td>
          Rs. 
          120
        </td>
        <td>
          1
        </td>
        <td>
          120
        </td>
      </tr>
    </table>
    <div
      class="total-value"
    >
      <div>
        total: 
        120
      </div>
      <button>
        Checkout
      </button>
    </div>
  </div>
</div>
`;

exports[`Cart should render 1`] = `
<div>
  <div
    class="cart-page"
  >
    <table
      class="cart-table"
    >
      <tr>
        <td>
          Item
        </td>
        <td>
          Unit Price
        </td>
        <td>
          Quatity
        </td>
        <td>
          Sub Total
        </td>
      </tr>
      <tr>
        <td>
          apple
        </td>
        <td>
          Rs. 
          120
        </td>
        <td>
          1
        </td>
        <td>
          120
        </td>
      </tr>
    </table>
    <div
      class="total-value"
    >
      <div>
        total: 
        120
      </div>
      <button>
        Checkout
      </button>
    </div>
  </div>
</div>
`;
